### PR TITLE
APPS-147 update to v2

### PR DESCRIPTION
**Pull request by mhrvol**
applet and app builder for xenial already published on staging. Tested on TTYD app(let) (ubuntu 16.04). After review I'll do the same on prod.

## App checklist

[Transition to Apps Doc page](https://documentation.dnanexus.com/developer/apps/transitioning-from-applets-to-apps)

### dxapp.json and Source code

- [x] Conform to [App(let) style guide](App(let)-Style-Guide.md).
- [ ] In `details` field specify: `whatsNew`, and `citation`.
- [ ] In `details` field, specify an array of `upstreamProjects` following App Style Guide [recommendations](App(let)-Style-Guide.md#licenses).
- [x] Verify `regionalOptions` contains regions for ALL supported app regions.
- [ ] Set applicable timeout and restart policy in [`runSpec`](https://documentation.dnanexus.com/developer/apps/app-metadata#runspec-entry-point-and-dependencies).
- [ ] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Contemplate setting `"openSource": true`. Default is `False`, set at your discretion.
- [ ] Add a unit test (or 2, or 3...) for your app. Check the App Test guide: [[latest](https://github.com/dnanexus/file-apps/blob/master/docs/notebook/writing_dxpy_test_scripts.ipynb)][[permalink](https://github.com/dnanexus/file-apps/blob/086e629ed6ea0f9d0f6ca7bd10c7f7027dff3443/docs/notebook/writing_dxpy_test_scripts.ipynb)]
- [ ] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.

### Documentation

- [ ] Branch of for Documentation
- [ ] Write Readme.md and (optional) Readme.developer.md
- [ ] Build as app on Production platform, bill to `org-dnanexus` (**DO NOT PUBLISH YET**). Remember unpublished app-xxxx
- [ ] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)
- [ ] Make a Pull Request to `master` in file-apps. Tag another scientist for review, link ubpublished app URL.

### Production

- [ ] Build as app on Production platform, bill to `org-dnanexus` (**DO NOT PUBLISH YET**). Remember unpublished app can be viewed on the platform, https://platform.dnanexus.com/app/app-xxxx
- [ ] Test it on expected data
- [ ] Verify data objects are for production and exist for all supported app regions in internally shared projects. Data Object projects are: Global search "App and Assets <Region>"
- [ ] Add Developers via [CLI call](https://documentation.dnanexus.com/user/helpstrings-of-sdk-command-line-utilities#add-developers). Make sure to add `org-dnanexus` as a developer so other scientist can view the unpublished app.
- [ ] Get PR approved.
- [ ] Add authorized users via [CLI call](https://documentation.dnanexus.com/user/helpstrings-of-sdk-command-line-utilities#add-users)
- [ ] Publish via [CLI call](https://documentation.dnanexus.com/user/helpstrings-of-sdk-command-line-utilities#publish)
- [ ] Verify the versions that appear on the platform are the intended versions

### CODE REVIEW

**Comment by commandlinegirl** [comment]
Oh, interesting, so the app doesn't have to be published in all regions to be able to build apps available in any region?

**Comment by asimenos** [comment]
It would have to. Not all user accounts will have billable access to US East.

File: tarball_app_builder_xenial/dxapp.json [hunk 1]
```diff
@@ -5,7 +5,16 @@
   "categories": ["Debugging"],
   "dxapi": "1.0.0",
   "billTo": "org-dnanexus",
-  "runSpec": {"file": "tarball_app_builder.py", "interpreter": "python2.7", "execDepends": [{"name": "build-essential"}], "distribution": "Ubuntu", "release": "16.04", "systemRequirements": {"*": {"instanceType": "mem1_ssd1_x2"}}},
+  "runSpec": {"file": "tarball_app_builder.py", "interpreter": "python2.7", "execDepends": [{"name": "build-essential"}], "distribution": "Ubuntu", "release": "16.04"},
+  "regionalOptions": {
+    "aws:us-east-1": {"systemRequirements": {"*": {"instanceType": "mem2_ssd1_v2_x4"}}},
```
Comment (commandlinegirl) [inline tarball_app_builder_xenial/dxapp.json:10]: LGTM, I'd only leave a cheaper instance type set as default, e.g. mem1_ssd1_v2_x2.

**Comment by bhuffman** [comment]
Could someone have an admin take me off of this repo?

**Comment by mhrvol** [comment]
> LGTM, modulo instance type update (to something cheaper by deafult, e.g. mem1_ssd1_v2_x4). Thanks!

is mem1_ssd1_v2_x2 ok as well? This is even cheaper.

File: tarball_app_builder_xenial/dxapp.json [hunk 2]
```diff
@@ -5,7 +5,16 @@
...
   "billTo": "org-dnanexus",
-  "runSpec": {"file": "tarball_app_builder.py", "interpreter": "python2.7", "execDepends": [{"name": "build-essential"}], "distribution": "Ubuntu", "release": "16.04", "systemRequirements": {"*": {"instanceType": "mem1_ssd1_x2"}}},
+  "runSpec": {"file": "tarball_app_builder.py", "interpreter": "python2.7", "execDepends": [{"name": "build-essential"}], "distribution": "Ubuntu", "release": "16.04"},
+  "regionalOptions": {
+    "aws:us-east-1": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_v2_x4"}}},
+    "aws:eu-west-2": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_v2_x4"}}},
+    "aws:eu-central-1": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_v2_x4"}}},
+    "aws:cn-north-1": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_v2_x4"}}},
+    "aws:ap-southeast-2": {"systemRequirements": {"*": {"instanceType": "mem1_ssd1_v2_x4"}}},
+    "azure:westus": {"systemRequirements": {"*": {"instanceType": "azure:mem2_ssd1_x4"}}},
```
Comment (commandlinegirl) [inline tarball_app_builder_xenial/dxapp.json:15]: Please update azure to a corresponding instance (mem1_ssd1_x4)

**Pull request merged**